hydra-http.c

```diff
@@ -426,54 +468,135 @@ int32_t service_http_init(char *ip, int32_t sp, unsigned char options, char *mis
   //   -1  error, hydra will exit, so print a good error message here
 
   /*POU CODE */
-  char *start = strstr(miscptr, "F=");
-  if (start == NULL)
-    start = strstr(miscptr, "S=");
-
-  if (start != NULL) {
-    if (start[0] == 'F')
-      end_condition_type = 0;
-    else
-      end_condition_type = 1;
-
-    int condition_len = strlen(start);
-    memset(end_condition, 0, END_CONDITION_MAX_LEN);
-    if (condition_len >= END_CONDITION_MAX_LEN) {
-      hydra_report(stderr, "Condition string cannot be bigger than %u.", END_CONDITION_MAX_LEN);
-      return -1;
+
+
+
+  char *misc = (char *)malloc(strlen(miscptr) + 1);
+  char *cp = misc;
+  memset(misc, '\0', strlen(miscptr));
+  strcpy(misc, miscptr);
+
+  char *delim = ":";
+  char *p = NULL;
+  char* match_text_start = NULL;
+  memset(match_status_code, '\0', MAX_STATUS_CODE_SIZE);
+
+#if defined __APPLE__ || __linux__ || __unix__
+  while ((p = strtok_r(NULL, delim, &misc))) {
+#endif
+
+#ifdef _WIN32
+    while ((p = strtok_s(NULL, delim, &misc))) {
+#endif
+
+      if (strstr(p, "r=") != NULL || strstr(p, "R=") != NULL) {
+        if (strlen(p) < 3) {
+          hydra_report(stderr, "Invalid status code, eg: 200 or 200|400.");
+          free(cp);
+          misc = NULL;
+          return -1;
+        }
+
+        char tmp;
+        int value, n;
+        int pos = 0;
+
+        p += 2;
+
+        while (p) {
+          n = sscanf(p, "%3d%c", &value, &tmp);
```

Comment:
you should remove the 3 of %3d, because otherwise you cannot match below if the value is larger than 1000